frontend/src/app/uploads/DeviceLookup/DeviceLookup.tsx

```diff
@@ -0,0 +1,261 @@
+import React, { useEffect, useState } from "react";
+import { ComboBox } from "@trussworks/react-uswds";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faCheck, faCopy } from "@fortawesome/free-solid-svg-icons";
+
+import TextInput from "../../commonComponents/TextInput";
+import { DeviceType } from "../../../generated/graphql";
+import LabeledText from "../../commonComponents/LabeledText";
+import Optional from "../../commonComponents/Optional";
+import "./DeviceLookup.scss";
+
+interface Device {
+  internalId?: string;
+  name: string;
+  manufacturer: string;
+  model: string;
+  loincCode: string;
+  swabTypes: Array<string>;
+}
+
+interface SwabType {
+  swabName: string;
+  typeCode: string;
+  internalId: string;
+}
+
+interface Props {
+  formTitle: string;
+  swabOptions: Array<SwabType>;
+  deviceOptions?: DeviceType[];
+}
+
+const DeviceLookup = (props: Props) => {
+  const [device, updateDevice] = useState<Device | undefined>();
+
+  const [copiedSlug, setCopiedSlug] = useState<string>();
+
+  useEffect(() => {
+    const timeout = copiedSlug
+      ? setTimeout(() => {
+          setCopiedSlug(undefined);
+        }, 3000)
+      : undefined;
+    return () => {
+      if (timeout) {
+        clearTimeout(timeout);
+      }
+    };
+  }, [copiedSlug]);
+
+  async function copySlug(slug: string) {
+    try {
+      await navigator.clipboard.writeText(slug);
+      setCopiedSlug(slug);
+    } catch (e: any) {
+      console.error(e);
+    }
+  }
+
+  const onChange = () => {};
+
+  const getDeviceOptions = () =>
+    props.deviceOptions
+      ? props.deviceOptions
+          .map((deviceType) => ({
+            label: deviceType.name,
+            value: deviceType.internalId,
+          }))
+          .sort((a, b) => a.label.localeCompare(b.label))
+      : [];
+
+  const getDeviceFromDeviceType = (device?: DeviceType): Device | undefined => {
+    return device
+      ? {
+          internalId: device.internalId,
+          name: device.name,
+          manufacturer: device.manufacturer,
+          model: device.model,
+          swabTypes: device.swabTypes?.map((swab) => swab.internalId),
+          loincCode: device.loincCode,
+        }
+      : undefined;
+  };
+
+  const getCopyToClipboardButton = (
+    copiedAttribute: string | undefined,
+    label: string
+  ) => {
+    return (
+      <div className="copy-button-container copy-button-tooltip">
+        {device && (
+          <button
+            className="usa-button usa-button--unstyled"
+            onClick={() => copiedAttribute && copySlug(copiedAttribute)}
+            aria-label={`${label}`}
+          >
+            <FontAwesomeIcon
+              icon={copiedSlug === copiedAttribute ? faCheck : faCopy}
+            />
+          </button>
+        )}
+        {device && copiedSlug === copiedAttribute && <CopyTooltip />}
+      </div>
+    );
+  };
+
+  return (
+    <main className="prime-home device-lookup-container">
```

Comment:
This is a duplicate `<main>` tag since the [`Page`](https://github.com/CDCgov/prime-simplereport/blob/main/frontend/src/app/commonComponents/Page/Page.tsx#L52) component wraps this component. This can be replaced with a div and just add `flex-1` to stretch the size.